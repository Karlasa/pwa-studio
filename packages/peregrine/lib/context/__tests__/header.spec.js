import React, { useEffect } from 'react';
import { createTestInstance } from '@magento/peregrine';

import HeaderContextProvider, { useHeaderContext } from '../header';

const log = jest.fn();
const Consumer = jest.fn(() => {
    const contextValue = useHeaderContext();

    const [{ headerRef }] = contextValue;

    useEffect(() => {
        log(contextValue);
    }, [contextValue]);

    return <i ref={headerRef} />;
});

test('renders children', () => {
    const symbol = Symbol();
    const tree = createTestInstance(
        <HeaderContextProvider>
            <i symbol={symbol} />
        </HeaderContextProvider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('provides state and actions via context', () => {
    createTestInstance(
        <HeaderContextProvider>
            <Consumer />
        </HeaderContextProvider>
    );

    expect(log.mock.calls[0]).toMatchSnapshot();
});
