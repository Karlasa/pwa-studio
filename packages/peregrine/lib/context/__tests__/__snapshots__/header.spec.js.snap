// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provides state and actions via context 1`] = `
Array [
  Array [
    Object {
      "headerRef": Object {
        "current": null,
      },
    },
    Object {
      "updateHeaderState": [Function],
    },
  ],
]
`;

exports[`renders children 1`] = `
<i
  symbol={Symbol()}
/>
`;
